### PR TITLE
refactor: logDeath

### PR DESCRIPTION
**Pull request by MafewTM**
## Description

Refactor the logging of deaths

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: client/dead.lua [hunk 1]
```diff
@@ -98,7 +97,8 @@ local function logDeath(victim, attacker, weapon)
     local killerName = GetPlayerName(killerId) .. ' ' .. '(' .. GetPlayerServerId(killerId) .. ')' or Lang:t('info.self_death')
     local weaponLabel = WEAPONS[weapon].label or 'Unknown'
     local weaponName = WEAPONS[weapon].name or 'Unknown'
-    TriggerServerEvent('qb-log:server:CreateLog', 'death', Lang:t('logs.death_log_title', { playername = playerName, playerid = GetPlayerServerId(playerId) }), 'red', Lang:t('logs.death_log_message', { killername = killerName, playername = playerName, weaponlabel = weaponLabel, weaponname = weaponName }))
+    local message = Lang:t('logs.death_log_message', { killername = killerName, playername = playerName, weaponlabel = weaponLabel, weaponname = weaponName })
+    lib.callback('qbx_medical:server:logDeath', false, false, message)
```
Comment (BerkieBb) [inline client/dead.lua:101]: Can just change it to await, no need for a vscode extension warning

**Review by BerkieBb — APPROVED** [review]

File: client/dead.lua (hunk 1 above)
Comment (MafewTM) [reply client/dead.lua:101]: What do you mean by that? Why should it be an await over a callback? And what is this warning you're talking about?

Comment (BerkieBb) [reply client/dead.lua:101]: If you send false to the argument which should be the callback function you get a warning in vscode for wrong data type, while not wrong, it's annoying

Comment (MafewTM) [reply client/dead.lua:101]: I get no warning regarding that when using the overextended lua extension and the sumneko lua extension and it doesnt error anything out in FiveM?

**Review by TonybynMp4 — APPROVED** [review]
LGTM

**Pull request merged**